components/plugin_marketplace/marketplace_list/navigation_row/navigation_row.jsx

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import PropTypes from 'prop-types';
+import {changeOpacity, makeStyleFromTheme} from 'mattermost-redux/utils/theme_utils';
+
+import NavigationButton from './navigation_button';
+
+export default class NavigationRow extends React.PureComponent {
+    static propTypes = {
+        page: PropTypes.number.isRequired,
+        total: PropTypes.number.isRequired,
+        maximumPerPage: PropTypes.number.isRequired,
+        onNextPageButtonClick: PropTypes.func.isRequired,
+        onPreviousPageButtonClick: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+    };
+
+    renderNextButton = () => {
+        const {page, maximumPerPage, total} = this.props;
+        const pageStart = page * maximumPerPage;
+        const pageEnd = pageStart + maximumPerPage;
+        const show = total >= maximumPerPage && pageEnd < total;
+
+        return show ? (
+            <NavigationButton
+                onClick={this.props.onNextPageButtonClick}
+                messageId={'more_channels.next'}
+                defaultMessage={'Next'}
+            />
+        ) : null;
+    };
+
+    renderPreviousButton = () => {
+        return this.props.page > 0 ? (
+            <NavigationButton
+                onClick={this.props.onPreviousPageButtonClick}
+                messageId={'more_channels.prev'}
+                defaultMessage={'Previous'}
+            />
+        ) : null;
+    };
+
+    renderCount = () => {
+        const {page, total, maximumPerPage} = this.props;
+        const startCount = page * maximumPerPage;
+        const endCount = Math.min(startCount + maximumPerPage, total);
+
+        return (
+            <FormattedMessage
+                id='marketplace_list.count_total_page'
+                defaultMessage='{startCount, number} - {endCount, number} {total, plural, one {plugin} other {plugins}} of {total, number} total'
+                values={{
+                    startCount: startCount + 1,
+                    endCount,
+                    total,
+                }}
+            />
+        );
+    };
+
+    render() {
+        const style = getStyle(this.props.theme);
+
+        return (
+            <div className='navigation-row'>
+                <div className='col-xs-2'>
+                    {this.renderPreviousButton()}
```

Comment:
there's a trick you can use to show the component which is, as opposed to calling a function return a component. You can:

```
{
	(this.props.page > 0) &&
	<NavigationButton
		onClick={this.props.onNextPageButtonClick}
        messageId={'more_channels.next'}
        defaultMessage={'Next'}
	/>
}
```

this way you don't have to do `return this.props.page > 0 ? comp : null;` react just hides that component with that condition is false.

same as next button 
```
{
	this.canShowNextButton() &&
		<NavigationButton
        	onClick={this.props.onNextPageButtonClick}
            messageId={'more_channels.next'}
            defaultMessage={'Next'}
        />
}
```

The reason this is preferred is because it's declarative and easier to reason about.

Reply:
Good point! Done :+1: